eftest/project.clj

```diff
@@ -7,4 +7,5 @@
                  [org.clojure/tools.namespace "0.2.11"]
                  [progrock "0.1.2"]
                  [io.aviso/pretty "0.1.33"]
-                 [mvxcvi/puget "1.0.1"]])
+                 [mvxcvi/puget "1.0.1"]
+                 [org.javassist/javassist "3.21.0-GA"]])
```

Comment:
The problem is the global java.lang.System/out and err. First, they are global and thus shared with all threads. Second, their type is java.io.PrintStream which is a concrete class, not an interface. And, third, it's always buffered.

First and third means that we can't share the PrintStream with multiple threads and still be able to separate what was printed by different threads.

So I came to a conclusion that I need an object that is PrintStream, but would allow capturing the calls to it and reroute the calls to different PrintStreams based on which thread is calling it.

Second point means that standard clojure.lang/proxy and java.lang.reflect.Proxy can't help, byte code generation is required. I know GCLIB, ASM, or Javassist, and I decided to use Javassist, mainly because I'm most comfortable with it.

So the reason for the this dependency is to allow creating an object that "is" PrintStream but allows me to reroute the method invocations.